ampd/src/mvx/verifier.rs

```diff
@@ -0,0 +1,608 @@
+use crate::handlers::mvx_verify_msg::Message;
+use crate::handlers::mvx_verify_verifier_set::VerifierSetConfirmation;
+use crate::types::Hash;
+use axelar_wasm_std::voting::Vote;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use multiversx_sdk::data::address::Address;
+use multiversx_sdk::data::transaction::{Events, TransactionOnNetwork};
+use hex::ToHex;
+use crate::mvx::WeightedSigners;
+
+const CONTRACT_CALL_IDENTIFIER: &str = "callContract";
+const CONTRACT_CALL_EVENT: &str = "contract_call_event";
+
+const ROTATE_SIGNERS_IDENTIFIER: &str = "rotateSigners";
+const SIGNERS_ROTATED_EVENT: &str = "signers_rotated_event";
+
+macro_rules! unwrap_or_continue {
+    ( $data:expr ) => {
+        match $data {
+            Some(x) => x,
+            None => continue,
+        }
+    };
+}
+
+impl Message {
+    fn eq_event(&self, event: &Events) -> Result<bool, Box<dyn std::error::Error>> {
+        if event.identifier != CONTRACT_CALL_IDENTIFIER {
+            return Ok(false);
+        }
+
+        let topics = event.topics.as_ref().ok_or("")?;
+
+        let event_name = topics.get(0).ok_or("")?;
+        let event_name = STANDARD.decode(event_name)?;
+        if event_name.as_slice() != CONTRACT_CALL_EVENT.as_bytes() {
+            return Ok(false);
+        }
+
+        let sender = topics.get(1).ok_or("")?;
+        let sender = STANDARD.decode(sender)?;
+        if sender.len() != 32 || &sender[0..32] != &self.source_address.to_bytes() {
+            return Ok(false);
+        }
+
+        let destination_chain = topics.get(2).ok_or("")?;
+        let destination_chain = STANDARD.decode(destination_chain)?;
+        let destination_chain = String::from_utf8(destination_chain)?;
+        if destination_chain != self.destination_chain.to_string() {
+            return Ok(false);
+        }
+
+        let destination_address = topics.get(3).ok_or("")?;
+        let destination_address = STANDARD.decode(destination_address)?;
+        let destination_address = String::from_utf8(destination_address)?;
+        if destination_address != self.destination_address {
+            return Ok(false);
+        }
+
+        let payload_hash = topics.get(4).ok_or("")?;

```

Comment:
Added a custom error, although these errors are ignored, since it will error if the data we need is not available, hence the message verification will fail (vote not found).